src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkBatchInsertRecord.java

```diff
@@ -193,7 +196,7 @@ else if (dateTimeFormatter != null)
                  * If the data is already a string, return it as is.
                  */
                 if (data instanceof byte[]) {
-                    return new String((byte[]) data);
+                    return new String((byte[]) data, Charset.forName(encoding));
```

Comment:
I would tweak the changes in this file so that we don't have to do this Charset lookup. It would be more efficient to:

1. Change the `String encoding` parameter to `Charset charset` and save it (replace the encoding member variable with charset).
2. The reference on line 52 changes from `encoding` to `charset.name()`. (We will only ever do this if logging is enabled.)
3. This line simply becomes `return new String((byte[]) data, charset);`
4. The parameter in constructor calls to SQLServerBulkBatchInsertRecord in SQLServerPreparedStatement simplifies from `connection.getDatabaseCollation().getCharset().name()` to just `connection.getDatabaseCollation().getCharset()`.